### ISSUE TITLE
No table are read in Starrocks 

### ISSUE THREAD

**Issue by moweonlee** [metabase/metabase#40433]
### Describe the bug

I have created connection to Starrocks 3.2.3 and Metabase 49.x, 48.x
They all failed to read tables in Starrocks. 
But in Starrocks 3.2.2, Metabase read data normally.

I'm not sure it could be Metabase's bug or Starrocks's bug.

### To Reproduce

1. Add database connection to  Starrocks 3.2.3 
2. No tables are read even I have clicked 'Sync database schema now' several times.

### Expected behavior

It should read all table.

### Logs

![image](https://github.com/metabase/metabase/assets/16459517/c514d90d-b89d-4432-b6a2-b5ef492982bd)

```
2024-03-21 18:54:28,563 WARN driver.mysql :: 

********************************************************************************
WARNING: Metabase only officially supports MySQL 5.7/MariaDB 10.2 and above.
All Metabase features may not work properly when using an unsupported version.
********************************************************************************

2024-03-21 18:54:28,564 INFO sync.util :: STARTING: Sync metadata for mysql Database 2 ''Starrocks''
2024-03-21 18:54:28,564 DEBUG middleware.log :: POST /api/database/2/sync_schema 200 26.6 ms (3 DB calls) App DB connections: 1/7 Jetty threads: 8/50 (8 idle, 0 queued) (51 total active threads) Queries in flight: 0 (0 queued)
2024-03-21 18:54:28,623 INFO sync.util :: STARTING: step ''sync-dbms-version'' for mysql Database 2 ''Starrocks''
2024-03-21 18:54:28,626 INFO sync.util :: FINISHED: step ''sync-dbms-version'' for mysql Database 2 ''Starrocks'' (2.6 ms)
2024-03-21 18:54:28,626 INFO sync.util :: STARTING: step ''sync-timezone'' for mysql Database 2 ''Starrocks''
2024-03-21 18:54:28,629 WARN sync.util :: Error running step ''sync-timezone'' for mysql Database 2 ''Starrocks''
java.sql.SQLTransientConnectionException: (conn=89) Getting analyzing error from line 1, column 73 to line 1, column 173. Detail message: No matching function with signature: time_format(TIME, varchar).
        at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.createException(ExceptionFactory.java:79)
        at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.create(ExceptionFactory.java:158)
        at org.mariadb.jdbc.MariaDbStatement.executeExceptionEpilogue(MariaDbStatement.java:262)
        at org.mariadb.jdbc.ClientSidePreparedStatement.executeInternal(ClientSidePreparedStatement.java:229)
        at org.mariadb.jdbc.ClientSidePreparedStatement.execute(ClientSidePreparedStatement.java:149)
        at org.mariadb.jdbc.ClientSidePreparedStatement.executeQuery(ClientSidePreparedStatement.java:163)
        at com.mchange.v2.c3p0.impl.NewProxyPreparedStatement.executeQuery(NewProxyPreparedStatement.java:1471)
        at clojure.java.jdbc$execute_query_with_params.invokeStatic(jdbc.clj:1090)
        at clojure.java.jdbc$execute_query_with_params.invoke(jdbc.clj:1084)
        at clojure.java.jdbc$db_query_with_resultset_STAR_.invokeStatic(jdbc.clj:1113)
        at clojure.java.jdbc$db_query_with_resultset_STAR_.invoke(jdbc.clj:1093)
        at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1182)
        at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
        at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1160)
        at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
        at metabase.driver.mysql$fn__103596.invokeStatic(mysql.clj:232)
        at metabase.driver.mysql$fn__103596.invoke(mysql.clj:222)
        at clojure.lang.MultiFn.invoke(MultiFn.java:234)
        at metabase.driver.sql_jdbc$fn__106405.invokeStatic(sql_jdbc.clj:72)
        at metabase.driver.sql_jdbc$fn__106405.invoke(sql_jdbc.clj:67)
        at clojure.lang.MultiFn.invoke(MultiFn.java:234)
        at metabase.sync.sync_metadata.sync_timezone$sync_timezone_BANG_.invokeStatic(sync_timezone.clj:41)
        at metabase.sync.sync_metadata.sync_timezone$sync_timezone_BANG_.invoke(sync_timezone.clj:36)
        at clojure.lang.AFn.applyToHelper(AFn.java:154)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.core$apply.invokeStatic(core.clj:669)
        at clojure.core$apply.invoke(core.clj:662)
        at metabase.sync.util$run_step_with_metadata$fn__60171.doInvoke(util.clj:462)
        at clojure.lang.RestFn.invoke(RestFn.java:397)
        at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:131)
        at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:125)
        at metabase.sync.util$with_start_and_finish_debug_logging.invokeStatic(util.clj:149)
        at metabase.sync.util$with_start_and_finish_debug_logging.invoke(util.clj:145)
        at metabase.sync.util$run_step_with_metadata.invokeStatic(util.clj:457)
        at metabase.sync.util$run_step_with_metadata.invoke(util.clj:452)
        at metabase.sync.util$run_sync_operation$fn__60250.invoke(util.clj:568)
        at metabase.sync.util$run_sync_operation.invokeStatic(util.clj:566)
        at metabase.sync.util$run_sync_operation.invoke(util.clj:560)
        at metabase.sync.sync_metadata$sync_db_metadata_BANG_$fn__82158.invoke(sync_metadata.clj:70)
        at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:190)
        at metabase.sync.util$do_with_error_handling.invoke(util.clj:183)
        at clojure.core$partial$fn__5910.invoke(core.clj:2647)
        at metabase.driver$fn__49757.invokeStatic(driver.clj:778)
        at metabase.driver$fn__49757.invoke(driver.clj:778)
        at clojure.lang.MultiFn.invoke(MultiFn.java:239)
        at metabase.sync.util$sync_in_context$fn__60085.invoke(util.clj:166)
        at metabase.sync.util$with_db_logging_disabled$fn__60082.invoke(util.clj:158)
        at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:131)
        at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:125)
        at metabase.sync.util$with_start_and_finish_logging$fn__60069.invoke(util.clj:143)
        at metabase.sync.util$with_sync_events$fn__60064.invoke(util.clj:117)
        at metabase.sync.util$with_duplicate_ops_prevented$fn__60051.invoke(util.clj:89)
        at metabase.sync.util$do_sync_operation.invokeStatic(util.clj:215)
        at metabase.sync.util$do_sync_operation.invoke(util.clj:209)
        at metabase.sync.sync_metadata$sync_db_metadata_BANG_.invokeStatic(sync_metadata.clj:68)
        at metabase.sync.sync_metadata$sync_db_metadata_BANG_.invoke(sync_metadata.clj:65)
        at metabase.api.database$fn__98315$fn__98320.invoke(database.clj:1010)
        at clojure.core$binding_conveyor_fn$fn__5823.invoke(core.clj:2047)
        at clojure.lang.AFn.call(AFn.java:18)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException: Getting analyzing error from line 1, column 73 to line 1, column 173. Detail message: No matching function with signature: time_format(TIME, varchar).
        at org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException.of(MariaDbSqlException.java:34)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:195)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:178)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:322)
        at org.mariadb.jdbc.ClientSidePreparedStatement.executeInternal(ClientSidePreparedStatement.java:220)
        ... 59 more
Caused by: java.sql.SQLException: Getting analyzing error from line 1, column 73 to line 1, column 173. Detail message: No matching function with signature: time_format(TIME, varchar).
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readErrorPacket(AbstractQueryProtocol.java:1693)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readPacket(AbstractQueryProtocol.java:1555)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.getResult(AbstractQueryProtocol.java:1518)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:319)
        ... 60 more
2024-03-21 18:54:28,629 INFO sync.util :: FINISHED: step ''sync-timezone'' for mysql Database 2 ''Starrocks'' (3.0 ms)
2024-03-21 18:54:28,630 INFO sync.util :: STARTING: step ''sync-tables'' for mysql Database 2 ''Starrocks''
2024-03-21 18:54:28,632 INFO sync.util :: FINISHED: step ''sync-tables'' for mysql Database 2 ''Starrocks'' (2.5 ms)
2024-03-21 18:54:28,632 INFO sync.util :: STARTING: step ''sync-fields'' for mysql Database 2 ''Starrocks''
2024-03-21 18:54:28,633 INFO sync.util :: FINISHED: step ''sync-fields'' for mysql Database 2 ''Starrocks'' (1.1 ms)
2024-03-21 18:54:28,633 INFO sync.util :: STARTING: step ''sync-fks'' for mysql Database 2 ''Starrocks''
2024-03-21 18:54:28,636 INFO sync.util :: FINISHED: step ''sync-fks'' for mysql Database 2 ''Starrocks'' (2.1 ms)
2024-03-21 18:54:28,636 INFO sync.util :: STARTING: step ''sync-indexes'' for mysql Database 2 ''Starrocks''
2024-03-21 18:54:28,637 INFO sync.util :: FINISHED: step ''sync-indexes'' for mysql Database 2 ''Starrocks'' (907.4 μs)
2024-03-21 18:54:28,637 INFO sync.util :: STARTING: step ''sync-metabase-metadata'' for mysql Database 2 ''Starrocks''
2024-03-21 18:54:28,637 INFO sync.util :: FINISHED: step ''sync-metabase-metadata'' for mysql Database 2 ''Starrocks'' (78.2 μs)
2024-03-21 18:54:28,637 INFO sync.util :: STARTING: step ''sync-table-privileges'' for mysql Database 2 ''Starrocks''
2024-03-21 18:54:28,637 INFO sync.util :: FINISHED: step ''sync-table-privileges'' for mysql Database 2 ''Starrocks'' (81.7 μs)
2024-03-21 18:54:28,648 INFO sync.util :: FINISHED: Sync metadata for mysql Database 2 ''Starrocks'' (83.9 ms)
2024-03-21 18:54:28,649 INFO sync.util :: STARTING: Analyze data for mysql Database 2 ''Starrocks''
2024-03-21 18:54:28,650 INFO sync.util :: STARTING: step ''fingerprint-fields'' for mysql Database 2 ''Starrocks''
2024-03-21 18:54:28,652 INFO sync.util :: FINISHED: step ''fingerprint-fields'' for mysql Database 2 ''Starrocks'' (1.9 ms)
2024-03-21 18:54:28,652 INFO sync.util :: STARTING: step ''classify-fields'' for mysql Database 2 ''Starrocks''
2024-03-21 18:54:28,652 INFO sync.util :: FINISHED: step ''classify-fields'' for mysql Database 2 ''Starrocks'' (62.9 μs)
2024-03-21 18:54:28,652 INFO sync.util :: STARTING: step ''classify-tables'' for mysql Database 2 ''Starrocks''
2024-03-21 18:54:28,652 INFO sync.util :: FINISHED: step ''classify-tables'' for mysql Database 2 ''Starrocks'' (72.4 μs)
2024-03-21 18:54:28,664 INFO sync.util :: FINISHED: Analyze data for mysql Database 2 ''Starrocks'' (8.3 ms)

```

![image](https://github.com/metabase/metabase/assets/16459517/347fcf28-9e4f-485a-8397-3be81766918b)

### Information about your Metabase installation

```JSON
- Metabase version : v0.49.1
- OS : Red Hat Enterprise Linux release 8.8 (Ootpa)
- Installation : Jar file downloaded
- Chrome browser
- Starrocks version 3.2.3
```

### Severity

No starrocks table can be used.

**Comment by paoliniluis** [comment]
If Metabase can’t use a function that should be there then we can’t read the tables. Seems you already put this on the starrocks GitHub issue tracker.

if starrocks is different from Mariadb or MySQL we can’t do nothing https://github.com/StarRocks/starrocks/issues/41951

**Comment by moweonlee** [comment]
@paoliniluis 

> If Metabase can’t use a function that should be there then we can’t read the tables.

 In Starrocks 3.2.2, I was able to read tables even though there were same time_format() function missing error. 
So I thought there are another issues behind it.

I started to see source code from now.
Do  you think Starrocks 3.2.3 can read data if I replace missing function with alternatives that Starrocks supports ?

I need Metabase + Starrocks :)
